0008-S-sidecar-endpoint-tls.md

```diff
@@ -0,0 +1,118 @@
+2. Add TLS support only, giving each SDK room to decide on how to expose it to the user
+  - Not every SDK offers an environment variable to configure Dapr endpoint, forcing configuration in code
+  - Environment variables per SDK, forcing sysadmin to know about each application's language use
+
+* Are there any trade-offs being made? (space for time, for example)
+1. Leaving existing environment variables for host and port as-is per SDK, but driving consistency on this new way.
+2. Not changing Dapr's DAPR_HOST (or equivalent), DAPR_HTTP_PORT and DAPR_GRPC_PORT.
+
+* What advantages / disadvantages does this proposal have? 
+Pros:
+- Bring consistency in Dapr API endpoint configuration cross SDKs
+- Add support for TLS endpoint
+
+Cons:
+- Does not address existing inconsistencies in client instantiation and env variables
+- Needs to define a priority between new env variables and old ones
+
+## Implementation Details
+
+### Design
+
+* `DAPR_GRPC_ENDPOINT` defines entire endpoit for gRPC, not just host: `https://dapr-grpc.mycompany.com`
+* `DAPR_HTTP_ENDPOINT` defines entire endpoit for HTTP, not just host: `https://dapr-http.mycompany.com`
+* Port is parsed from the URL (`https://dapr.mycompany.com:8080`) or via the default port of the protocol used in the URL (80 for `http` and 443 for `https`)
+* `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT` can be set at the same time since some SDKs (Java, as of now) supports both protocols at the same time and app can pick which one to use.
+* `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT` must be parsed and the protocol will be used for SDK to determine if communication is over TLS (if not done automatically). In summary, `https` means secure channel.
+* Initially, only `http` and `https` protocols should be supported. Future protocols can be added in the future depending on each language support.
+* `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT` have priority over existing `DAPR_HOST` and `DAPR_HTTP_PORT` or `DAPR_GRPC_PORT` environment variables. Application's hardcoded values passed via constructor takes priority over any environment variable. In summary, this is the priority list (highest on top):
+  1. Values passed via constructor or builder method.
+  2. `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT`
+  3. Existing `DAPR_HOST` (or equivalent, defaulting to `127.0.0.1`) + `DAPR_HTTP_PORT` or `DAPR_GRPC_PORT`
+
+#### Example of implementation
+
+https://github.com/dapr/java-sdk/blob/76aec01e9aa4af7a72b910d77685ddd3f0bf86f3/sdk/src/main/java/io/dapr/client/DaprClientBuilder.java#L172C3-L192
+
+### Feature lifecycle outline
+
+* Compatability guarantees
+This feature should allow localhost definition too `http://127.0.0.1:3500`, for example.
+
+* Deprecation / co-existence with existing functionality
+This feature takes priority over existing (inconsistent) environment variables from each SDK. If app provides a hardcoded value for Dapr endpoint (via constructor, for example), it takes priority.
```

Comment:
This is a key point that I really like, the fact the existing programmatic interfaces are unchanged and taking precedence over the environment variables retains backward compatibility fully.